src/coreclr/gc/gc.cpp

```diff
@@ -23543,14 +23540,21 @@ void gc_heap::save_post_plug_info (uint8_t* last_pinned_plug, uint8_t* last_obje
     }
 }
 
-//#define PREFETCH
+#define PREFETCH
 #ifdef PREFETCH
-__declspec(naked) void __fastcall Prefetch(void* addr)
+inline void Prefetch(void* addr)
 {
-   __asm {
-       PREFETCHT0 [ECX]
-        ret
-    };
+#ifdef TARGET_AMD64
+
+#ifndef _MM_HINT_T0
+#define _MM_HINT_T0 1
+#endif
+    _mm_prefetch((const char*)addr, _MM_HINT_T0);
+#elif defined(TARGET_ARM64) && defined(TARGET_WINDOWS)
+    __prefetch((const char*)addr);
```

Comment:
`__builtin_prefetch` should work on non-Windows

Reply:
https://clang.llvm.org/docs/LanguageExtensions.html describes the arguments.

Reply:
ah right, on linux we should use `__buildin_prefetch`.

Reply:
Thanks Jan - it looks like calling it with the default arguments should be just fine for our purposes.